python/cudf/cudf/core/column/column.py

```diff
@@ -1197,7 +1197,15 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
 
     elif isinstance(arbitrary, Buffer):
         if dtype is None:
-            raise TypeError("dtype cannot be None if 'arbitrary' is a Buffer")
+            # If the Buffer/Serializable object has
+            # __cuda_array_interface__ implemented, pick dtype from it.
+            if hasattr(arbitrary, "__cuda_array_interface__"):
+                dtype = np.dtype(arbitrary.__cuda_array_interface__["typestr"])
```

Comment:
Doesn't the base class of `Buffer` expose `__cuda_array_interface__` as well?

Reply:
It does, maybe we should stop throwing error if the `dtype` is not passed?

Reply:
I think the whole point of this is to error if a `Buffer` object is passed without dtype. Maybe instead of using `isinstance(arbitrary, Buffer)` we check explicitly `type(arbitrary) is Buffer`.

Reply:
Changed the check :+1: